fe/fe-core/src/main/java/com/starrocks/persist/DropPartitionsInfo.java

```diff
@@ -0,0 +1,133 @@
+
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package com.starrocks.persist;
+
+import com.google.common.base.Objects;
+import com.google.gson.annotations.SerializedName;
+import com.starrocks.common.io.Text;
+import com.starrocks.common.io.Writable;
+import com.starrocks.persist.gson.GsonUtils;
+
+import java.io.DataInput;
+import java.io.DataOutput;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+public class DropPartitionsInfo implements Writable {
+    @SerializedName(value = "dbId")
+    private Long dbId;
+    @SerializedName(value = "tableId")
+    private Long tableId;
+    @SerializedName(value = "isTempPartition")
+    private boolean isTempPartition = false;
+    @SerializedName(value = "forceDrop")
+    private boolean forceDrop = false;
+    @SerializedName(value = "partitionNames")
+    private List<String> partitionNames = new ArrayList<>();
+
+    private DropPartitionsInfo() {
+    }
+
+    public DropPartitionsInfo(Long dbId, Long tableId, boolean isTempPartition, boolean forceDrop, List<String> partitionNames) {
+        this.dbId = dbId;
+        this.tableId = tableId;
+        this.isTempPartition = isTempPartition;
+        this.forceDrop = forceDrop;
+        this.partitionNames = partitionNames;
+    }
+
+    public Long getDbId() {
+        return dbId;
+    }
+
+    public Long getTableId() {
+        return tableId;
+    }
+
+    public boolean isTempPartition() {
+        return isTempPartition;
+    }
+
+    public boolean isForceDrop() {
+        return forceDrop;
+    }
+
+    public List<String> getPartitionNames() {
+        return partitionNames;
+    }
+
+    public void setPartitionNames(List<String> partitionNames) {
+        this.partitionNames = partitionNames;
+    }
+
+    public static DropPartitionsInfo read(DataInput in) throws IOException {
+        String json = Text.readString(in);
+        return GsonUtils.GSON.fromJson(json, DropPartitionsInfo.class);
+    }
+
+    @Override
+    public void write(DataOutput out) throws IOException {
+        String json = GsonUtils.GSON.toJson(this);
+        Text.writeString(out, json);
+    }
+
```

Comment:
No need to implement the hasCode() and equals() interface, do you known where to use it?

Reply:
Yes, currently there doesn't seem to be any necessary usage of them. I have removed these two methods.

Reply:
The DropPartitionsInfoTest.testSerialization method will be used inside the assertion.